src/main/java/eu/dissco/core/provenanceservice/service/ProcessingService.java

```diff
@@ -18,47 +17,44 @@
 @RequiredArgsConstructor
 public class ProcessingService {
 
+  private static final Map<String, String> SUBJECT_MAPPING = provideSubjectMapping();
   private final ObjectMapper mapper;
   private final EventRepository eventRepository;
-  private static final Map<String, String> SUBJECT_MAPPING = provideSubjectMapping();
 
   private static Map<String, String> provideSubjectMapping() {
     var map = new HashMap<String, String>();
-    map.put("DigitalSpecimen", "digital_specimen_provenance");
-    map.put("DigitalMediaObject", "digital_media_provenance");
-    map.put("Annotation", "annotation_provenance");
-    map.put("MachineAnnotationService", "machine_annotation_service_provenance");
-    map.put("Mapping", "mapping_provenance");
-    map.put("SourceSystem", "source_system_provenance");
+    map.put("ods:DigitalSpecimen", "digital_specimen_provenance");
+    map.put("ods:DigitalMedia", "digital_media_provenance");
+    map.put("ods:Annotation", "annotation_provenance");
+    map.put("ods:MachineAnnotationService", "machine_annotation_service_provenance");
+    map.put("ods:DataMapping", "data_mapping_provenance");
+    map.put("ods:SourceSystem", "source_system_provenance");
     return map;
   }
 
   public void handleMessage(String message)
       throws JsonProcessingException, MongodbException, UnknownSubjectException {
-    var event = mapper.readValue(message, CreateUpdateDeleteEvent.class);
-    var versionId = generateUniqueVersionId(event.eventRecord());
-    var collectionName = parseSubjectType(event.subjectType());
+    var event = mapper.readValue(message, CreateUpdateTombstoneEvent.class);
+    var versionId = event.getId();
+    var collectionName = parseSubjectType(event);
     var eventResult = eventRepository.insertNewVersion(versionId, event, collectionName);
     if (eventResult) {
-      log.info("Successfully processed event information for {}: {}", event.subjectType(),
+      log.info("Successfully processed event information for {}: {}",
+          event.getProvEntity().getType(),
           versionId);
     } else {
       log.warn("Failed to insert event into mongodb: {}", message);
       throw new MongodbException(message);
     }
   }
 
-  private String parseSubjectType(String subjectType) throws UnknownSubjectException {
-    var collectionName = SUBJECT_MAPPING.get(subjectType);
-    if (collectionName == null){
-      throw new UnknownSubjectException("SubjectType: " + subjectType + " is unknown");
+  private String parseSubjectType(CreateUpdateTombstoneEvent event) throws UnknownSubjectException {
+    var collectionName = SUBJECT_MAPPING.get(event.getProvEntity().getType());
+    if (collectionName == null) {
+      throw new UnknownSubjectException("SubjectType: " + event + " is unknown");
```

Comment:
do we want to include the entire event in the exception, or just the subject?
I'd log the event, and include `SubjectType: " + event.getProvEntity().getType() + " is unknown"` in the exception

Reply:
Yes good point, changed it